core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -3158,14 +3158,14 @@ object ZIO extends ZIOCompanionPlatformSpecific with ZIOCompanionVersionSpecific
    * its identity.
    */
   def descriptorWith[R, E, A](f: Fiber.Descriptor => ZIO[R, E, A])(implicit trace: Trace): ZIO[R, E, A] =
-    ZIO.withFiberRuntime[R, E, A] { (fiberState, status) =>
+    ZIO.withFiberRuntime[R, E, A] { (fiberState, runtimeFlags) =>
       val descriptor =
         Fiber.Descriptor(
           fiberState.id,
-          status,
+          Fiber.Status.Running(runtimeFlags, trace),
```

Comment:
Previously, the `trace` was coming from the `Stateful` object built by the `ZIO.withFiberRuntime` but since this trace is the same as the implicit one passed in `descriptorWith`, if I'm correct, we can just pass the `trace` here and so only build the `Fiber.Status.Running` here instead of in the runloop.